dotnet/src/Connectors/Connectors.MistralAI/Client/MistralClient.cs

```diff
@@ -33,52 +34,89 @@ internal sealed class MistralClient
+        var request = this.CreateChatCompletionRequest(modelId, stream: false, chatHistory, mistralExecutionSettings);
+        var endpoint = this.GetEndpoint(mistralExecutionSettings, path: "chat/completions");
 
         using var httpRequestMessage = this.CreatePost(request, endpoint, this._apiKey, stream: false);
 
         var response = await this.SendRequestAsync<ChatCompletionResponse>(httpRequestMessage, cancellationToken).ConfigureAwait(false);
 
-        return response.Choices.Select(chatChoice => new ChatMessageContent(new AuthorRole(chatChoice.Message!.Role), chatChoice.Message!.Content, this._modelId, chatChoice, Encoding.UTF8, GetChatChoiceMetadata(response, chatChoice))).ToList();
+        return response.Choices.Select(chatChoice => new ChatMessageContent(new AuthorRole(chatChoice.Message!.Role!), chatChoice.Message!.Content, this._modelId, chatChoice, Encoding.UTF8, GetChatChoiceMetadata(response, chatChoice))).ToList();
     }
 
-    internal async IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageContentsAsync(ChatHistory chatHistory, CancellationToken cancellationToken, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null)
+    internal async IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageContentsAsync(ChatHistory chatHistory, [EnumeratorCancellation] CancellationToken cancellationToken, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null)
     {
         string modelId = executionSettings?.ModelId ?? this._modelId;
         var mistralExecutionSettings = MistralAIPromptExecutionSettings.FromExecutionSettings(executionSettings);
-        var request = this.CreateChatCompletionRequest(modelId, false, chatHistory, mistralExecutionSettings);
-        var endpoint = new Uri($"{this._endpoint}{this._separator}chat/completions");
+        var request = this.CreateChatCompletionRequest(modelId, stream: true, chatHistory, mistralExecutionSettings);
+        var endpoint = this.GetEndpoint(mistralExecutionSettings, path: "chat/completions");
 
         using var httpRequestMessage = this.CreatePost(request, endpoint, this._apiKey, stream: true);
 
         using var response = await this.SendStreamingRequestAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
 
         using var responseStream = await response.Content.ReadAsStreamAndTranslateExceptionAsync().ConfigureAwait(false);
-        foreach (var streamingTextContent in this.ProcessResponseStream(responseStream, modelId))
+        using var reader = new StreamReader(responseStream);
+        string line;
+        string? rawChunk = null;
+        AuthorRole? currentRole = null;
+        while ((line = await reader.ReadLineAsync().ConfigureAwait(false)) != null)
         {
-            yield return streamingTextContent;
+            if (!string.IsNullOrEmpty(line))
+            {
+                rawChunk = line.Substring("data:".Length).Trim();
```

Comment:
Use a constant for "data:".Length 
i.e.: Avoid allocation of a new string and getting the same length.

```suggestion
                rawChunk = line.Substring(SSEDataLength).Trim();
```